vault/resource_generic_secret_test.go

```diff
@@ -29,6 +28,31 @@ func TestResourceGenericSecret(t *testing.T) {
 	})
 }
 
+func TestResourceGenericSecret_deleted(t *testing.T) {
+	path := acctest.RandomWithPrefix("secret/test")
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testResourceGenericSecret_initialConfig(path),
+				Check:  testResourceGenericSecret_initialCheck(path),
+			},
+			resource.TestStep{
+				PreConfig: func() {
+					client := testProvider.Meta().(*api.Client)
+					_, err := client.Logical().Delete(path)
+					if err != nil {
+						t.Fatal(err)
```

Comment:
Some more context about why this is a fatal condition would be helpful for future debugging of failed tests.